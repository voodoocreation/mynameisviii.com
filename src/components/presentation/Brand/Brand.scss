@import "../../../scss/variables", "../../../scss/mixins";

.Brand {
  $size: 48px;

  display: flex;
  width: $size;
  height: $size;
  flex-shrink: 0;
  flex-grow: 0;
  align-items: center;

  &::before {
    @include bg-2();
    @include logo-name();

    width: 100%;
    height: 100%;
    content: "";
    background-size: cover;
  }

  span {
    @include hide-visually();
  }

  @include when-below($screen-xs-max) {
    margin-right: $spacing;
  }

  @include when-above($screen-sm-min) {
    $size: $banner-size-medium - $spacing-double;

    width: $size;
    height: $size;
    margin: $spacing auto -3px;
  }

  @include when-above($screen-md-min) {
    $size: $banner-size - $spacing-double;

    width: $size;
    height: $size;
  }
}
