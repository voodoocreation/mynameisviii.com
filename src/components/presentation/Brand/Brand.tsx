import * as React from "react";
import { FormattedMessage } from "react-intl";

import Link from "../Link/Link";

import logoName from "../../../../static/img/logo-name.svg";
import logoSymbol from "../../../../static/img/logo-symbol.svg";

const Brand: React.SFC<{}> = () => (
  <Link className="Brand" href="/">
    <FormattedMessage id="BRAND_NAME" />
    <svg
      {...logoName.attributes}
      dangerouslySetInnerHTML={{ __html: logoName.content }}
    />
    <svg
      {...logoSymbol.attributes}
      dangerouslySetInnerHTML={{ __html: logoSymbol.content }}
    />
  </Link>
);

export default Brand;
