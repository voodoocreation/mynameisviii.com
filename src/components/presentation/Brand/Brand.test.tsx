import { render } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";

import Brand from "./Brand";

const setup = (fn: any) => ({
  actual: fn(<Brand />)
});

describe("[presentation] <Brand />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);
    expect(toJson(actual)).toMatchSnapshot();
  });
});
