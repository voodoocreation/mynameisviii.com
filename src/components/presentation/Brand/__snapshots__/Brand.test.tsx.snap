// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand /> renders correctly 1`] = `
<a
  class="Brand"
  href="/"
>
  <span>
    My name is Viii: a painter of frequencies and thoughts
  </span>
  <svg />
  <svg />
</a>
`;
